middleware/opentracing.go

```diff
@@ -0,0 +1,41 @@
+package middleware
+
+import (
+	"net/http"
+
+	"github.com/go-chi/chi"
+	"github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/ext"
+)
+
+// OpenTracing provides a middleware for cross-service tracing support.
+func OpenTracing(next http.Handler) http.Handler {
```

Comment:
Would be nice to see an example of how this integrates with the datadog tracer API.

Reply:
I think this is all you have to do:

```go
// Start a Datadog tracer, optionally providing a set of options,
// returning an opentracing.Tracer which wraps it.
t := opentracer.New(tracer.WithAgentAddr("host:port"))

// Use it with the Opentracing API. The (already started) Datadog tracer
// may be used in parallel with the Opentracing API if desired.
opentracing.SetGlobalTracer(t)
```

https://github.com/DataDog/dd-trace-go/blob/v1/ddtrace/opentracer/example_test.go#L15-L23